app/services/merge_patient_service.rb

```diff
@@ -23,7 +23,9 @@ def merge
     touch_patient(merged_patient) if associations_updated?(merged_address, merged_phone_numbers, merged_business_ids)
 
     # Patient has been soft-deleted by the client, server should soft-delete the patient and their associated data.
-    discard_patient_data(merged_patient) if discarded_in_this_merge?(merged_patient, patient_before_merge)
+    if discarded_in_this_merge?(merged_patient, patient_before_merge)
+      discard_patient_data(merged_patient, deleted_reason: merged_patient.deleted_reason)
```

Comment:
Can move the `merged_patient.deleted_reason` inside `discard_patient_data`.

Reply:
Let's make `deleted_reason` an [enum](https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/Enum.html) on `Patient`.